### DISCUSSION TITLE
Nocturnal Flight Call and Daytime Stereo Microphone setup possible?

### DISCUSSION THREAD

**Discussion by Astrobirder** [mcguirepr89/BirdNET-Pi#572] [Ideas]
Hello,

At the suggestion of @[nilsp](https://github.com/nilspupils), I've started this new discussion to split it out from the @DD4WH Microphone thread.

I'm about to dive in and make a few microphones, per the instructions in https://github.com/mcguirepr89/BirdNET-Pi/discussions/39. A question--has anyone tried using one of these microphones mounted "normally" along with one mounted like is recommended for Nocturnal Flight Calls here: http://oldbird.org/mike_home-DIY.htm and then using the "big white" ugreen USB soundcard to have a single installation that does both day and night monitoring?

Does anyone know if the existing birdnet model includes any nocturnal flight call training?  @mcguirepr89 ?  

Any ideas or interest?

**Comment by mcguirepr89** [comment]
I know of at least one system that is dedicated to NFCs and BirdNET-Lite does not really identify those calls hardly at all. I suspect that NFCs have not been trained into the model, unfortunately -- though I suspect they will make their way into the new model if we request it.

**Reply by Astrobirder** [reply]
Thanks @mcguirepr89, I also am wondering about Insect-trained and Bat-trained models.  I have a couple contacts @cornell, but haven't asked about this and didn't know if it was Cornell or Chemnitz that was releasing that piece.  Do you know if Insect or Bat focused recognizers exist as well?

I'm also curious about having combination recognizers--i.e. Daytime Bird, NFC, Day & Night Insect and Heterodyned Bat all in one, but I expect the False Positive rate on such a system to potentially be very high.  You *MIGHT* be able to use time windows to reduce that a bit (kind of like the regional rejection with bird lists that is there now), but I'm not knowledgeable enough on Machine Learn to know.

**Comment by nilspupils** [comment]
Regarding detection of NFCs, have a look at this:

https://github.com/HaroldMills/Vesper

and this: 

https://github.com/BirdVox/birdvoxdetect

They might be difficult to integrate into one system with BirdNET-Pi though...

**Comment by DD4WH** [comment]
Hi, good idea!
My experience with NFCs is:
* some really do get recorded here with BirdNET-Lite: Redwing, Common Quail, Water Rail, Coot, Moorhen, Little Grebe . . .
* however, of course, I cannot quantify how many do NOT get recorded. My listening site is not at all on one of the large European migration routes

Using TWO mics (one for daytime birds, one for NFCs):
* this is not feasible with the current software version. 
* One would have to implement a software switch that swaps the channels for the classification at a given time adjustable by the user OR calculated by the BirdNET-Pi for dusk and dawn. Even if you record in Stereo, BirdNET-Pi only uses ONE channel to classify the birds (as far as I know).

BAT Classifiers:
* this is a total different story and should be well separated from bird and insect classification: you need a much higher sample rate (at least 384ksps, best 500ksps) in order to capture the high frequencies. So with the current hardware (audio USB adapters) it is NOT possible to make reliable ultrasound recordings, because they do not allow such high sampling frequencies in their ADCs
* Also: bat echolocation calls are evolutionarily adapted to the habitat of a bat. They did not evolve as species-specific calls (unlike most of the bird songs and calls). However, many bat species in Europe and North America CAN be identified more or less reliably, but in many cases one must be satisfied with a species group or genus (eg. Plecotus species, Myotis etc.).
* Heterodyned bat calls are NOT identifiable any more, because you loose the frequency information and -more important- the bandwidth. As an example: for a Myotis bat identification, you need a bandwidth of more than 100kHz (20 to 120kHz at least). You will ruin this, if you heterodyne the sound
* Bat classifiers work on full spectrum recordings or on zero crossing recordings, without any pitch shifting such as heterodyning
      
My suggestion would be to stay modest and start with the idea of having a timer in the BirdNET-Pi software that switches the mic channel source that is being used by the classifier. Then you can have a bucket microphone for NFCs at night and an omnidirectional mic in the day.

**Reply by Astrobirder** [reply]
@DD4WH Clearly I don't understand how stereo systems work with Birdnet-Pi!  I thought (from your prior explanations) that the two audio channels got mixed and treated as a single channel (and used for recognition) when you use the White Ugreen system.  Clearly that isn't the case.  I'm now confused as to why you recommended that unit at one time.  Was it just for its low noise performance?

**Reply by Astrobirder** [reply]
Regarding the bats, I know you've done work with them @DD4WH (yes, I've been looking at your other github repos :-) ).  I was afraid it would not be so simple.  I've only ever pulled 1 species out of a Northern Saw-whet owl banding net, and have no experience with detectors--I'd just like to do that as a hobby project!   Wouldn't having a known/fixed frequency shift still allow for knowledge of the original frequency?

I'm trying to translate from my amateur radio experience here and am envisioning a fixed LO.  Is this just not possible due to the wider range in ultrasonic frequencies by bats?  i.e. you have to use various LO frequencies to get various bat genus/species into the audible range?

All that aside, I agree with the staying modest and doing the mic channel switch :-).

**Reply by DD4WH** [reply]
I recommended the UGreen device, because it is the only low-cost USB audio interface with STEREO mic input. So it can record in Stereo (I have two ears, so I WANT Stereo audio to listen to :-). Of course, only one channel is processed by BirdNET.

Yes, its a complex topic.

Here is a good summary and a good explanation, why heterodyne detectors (using a LO) are very good for education, but no longer a professional tool to identify bat species (you need full spectrum recordings and subsequent processing/analysis for that):

https://ecoobs.com/comparison-detector-technologies/
https://www.bats.org.uk/about-bats/bat-detectors-1/heterodyne
